### PR TITLE
[Merged by Bors] - feat: cardinalities of free constructions and `Nonempty (CommRing α)`

### PR DESCRIPTION
**Pull request by eric-wieser**
[Zulip thread](https://leanprover.zulipchat.com/#narrow/channel/116395-maths/topic/Does.20every.20type.20permit.20a.20semiring.3F/near/479347521)

Co-authored-by: Daniel Weber <55664973+Command-Master@users.noreply.github.com>

---

- [x] depends on: #18363

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/SetTheory/Cardinal/Free.lean [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+This file shows that all the free constructions over `α` have cardinality `max #α ℵ₀`,
+and are thus infinite.
+
+Combined with the ring `Fin n` for the finite cases, this lets us show that there is a `CommRing` of
+any cardinality.
+-/
+
+universe u
+variable (α : Type u)
```
Comment (riccardobrasca) [inline Mathlib/SetTheory/Cardinal/Free.lean:23]: Can we use `Type*` here?

**Review by riccardobrasca — COMMENTED** [review]
There is a building error, but LGTM, thanks!

**Comment by eric-wieser** [comment]
> There is a building error,

Fixed

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**